common/djangoapps/third_party_auth/models.py

```diff
@@ -226,7 +236,14 @@ def display_for_login(self):
         Determines whether the provider ought to be shown as an option with
         which to authenticate on the login screen, registration screen, and elsewhere.
         """
-        return bool(self.enabled and self.accepts_logins and self.visible)
+        return bool(self.enabled_for_current_site and self.accepts_logins and self.visible)
+
+    @property
+    def enabled_for_current_site(self):
```

Comment:
I'm willing to abstract an `enabled_for_site` out of this, which `enabled_for_current_site` uses, but there are many places that specifically need to check if a ProviderConfig is enabled against the current site.

Should I keep `enabled_for_current_site` and have it call the `enabled_for_site` property (using `Site.objects.get_current()` to fetch the current Site, as you mentioned in comments below)?